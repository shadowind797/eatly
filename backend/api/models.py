from django.db import models
from django.contrib.auth.models import AbstractUser


class User(AbstractUser):
    username = models.CharField(max_length=120, unique=True)
    password = models.CharField(max_length=120)
    phone = models.CharField(max_length=120, blank=True)
    status = models.ForeignKey('Status', on_delete=models.PROTECT, default=5)
    is_banned = models.BooleanField(default=False)
    ban_reason = models.TextField(blank=True)
    passwd_change_link = models.CharField(max_length=120, blank=True)

    def __str__(self):
        return self.username


class Item(models.Model):
    title = models.CharField(max_length=120, unique=True)
    description = models.TextField()
    price = models.FloatField()
    photo = models.ImageField(upload_to='img')
    category = models.ForeignKey('Category', on_delete=models.SET_NULL, null=True, blank=True)
    restaurant = models.ForeignKey('Restaurant', on_delete=models.SET_NULL, null=True, blank=True)
    rating = models.FloatField()

    def __str__(self):
        return self.title


class Restaurant(models.Model):
    name = models.CharField(max_length=120, unique=True)
    address = models.CharField(max_length=120)
    rating = models.FloatField()
    image = models.ImageField(upload_to='img/rests', null=True)
    category = models.ForeignKey('RestaurantCat', on_delete=models.SET_NULL, null=True, blank=True)

    def __str__(self):
        return self.name


class Address(models.Model):
    house_address = models.CharField(max_length=120)
    entrance = models.CharField(max_length=120)
    floor = models.CharField(max_length=120)
    flat = models.CharField(max_length=120)
    owner = models.ForeignKey('User', on_delete=models.CASCADE)
    active = models.BooleanField(default=False)

    def __str__(self):
        return self.house_address


class Order(models.Model):
    user = models.ForeignKey('User', on_delete=models.SET_NULL, null=True)
    payment = models.ForeignKey('Payments', on_delete=models.SET_NULL, null=True)
    address = models.ForeignKey('Address', on_delete=models.SET_NULL, null=True)
    total = models.FloatField()
    status = models.ForeignKey('OrderStatus', on_delete=models.SET_NULL, null=True)
    restaurant = models.ForeignKey('Restaurant', on_delete=models.SET_NULL, null=True)
    coupon = models.ForeignKey('Coupon', on_delete=models.SET_NULL, null=True)
    comment = models.TextField(null=True, blank=True)

    def __str__(self):
        return self.user


class OrderStatus(models.Model):
    name = models.CharField(max_length=120, unique=True)


class Coupon(models.Model):
    title = models.CharField(max_length=120, unique=True)
    valid_from = models.DateField(auto_now_add=True)
    valid_to = models.DateField()
    is_valid = models.BooleanField(default=True)
    value = models.FloatField()
    times_activated = models.IntegerField(default=0)
    enabled_times = models.IntegerField()
    category = models.ForeignKey('CouponCategory', on_delete=models.CASCADE)

    def __str__(self):
        return self.title


class CouponCategory(models.Model):
    name = models.CharField(max_length=120, unique=True)

    def __str__(self):
        return self.name


class Payments(models.Model):
    number = models.CharField(max_length=120, unique=True)
    date_to = models.CharField(max_length=100)
    cvv = models.IntegerField()
    owner = models.ForeignKey('User', on_delete=models.CASCADE)
    active = models.BooleanField(default=False)
    name = models.CharField(max_length=120, default="")

    def __str__(self):
        return self.number


class CartItem(models.Model):
    item = models.ForeignKey('Item', on_delete=models.CASCADE)
    quantity = models.IntegerField(default=1)
    owner = models.ForeignKey('User', on_delete=models.CASCADE)

    def __str__(self):
        return self.item.title


class Reviews(models.Model):
    author = models.ForeignKey('User', on_delete=models.CASCADE)
    time = models.CharField(max_length=120)
    text = models.TextField()
    rating = models.FloatField()
    created_at = models.DateTimeField(auto_now_add=True)

    def __str__(self):
        return self.author.username


class RestaurantCat(models.Model):
    name = models.CharField(max_length=120, unique=True)

    def __str__(self):
        return self.name


class Category(models.Model):
    name = models.CharField(max_length=120, unique=True)

    def __str__(self):
        return self.name


class Status(models.Model):
    name = models.CharField(max_length=120)

    def __str__(self):
        return self.name
