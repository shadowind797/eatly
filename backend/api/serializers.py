from .models import *
from rest_framework import serializers


class UserSerializer(serializers.ModelSerializer):
    class Meta:
        model = User
        fields = ['id', 'username', 'email', 'password', "phone", "status", "is_banned", "ban_reason", "first_name"]
        extra_kwargs = {"password": {"write_only": True}, "status": {"read_only": True}}

    def create(self, validated_data):
        user = User.objects.create_user(**validated_data)
        return user


class AddressSerializer(serializers.ModelSerializer):
    class Meta:
        model = Address
        fields = ["id", "house_address", "entrance", "floor", "flat"]

    def create(self, validated_data):
        address = Address.objects.create_address(**validated_data)
        return address


class PaymentSerializer(serializers.ModelSerializer):
    class Meta:
        model = Payments
        fields = "__all__"
        extra_kwargs = {"cvv": {"write_only": True}, "date_to": {"read_only": True}}

    def create(self, validated_data):
        payment = Payments.objects.create_payments(**validated_data)
        return payment


class OrderSerializer(serializers.ModelSerializer):
    class Meta:
        model = Order
        fields = ["id", "status", "total", "payment", "address"]

    def create(self, validated_data):
        order = Order.objects.create_order(**validated_data)
        return order


class ItemSerializer(serializers.ModelSerializer):
    class Meta:
        model = Item
        fields = ['id', "title", "description", "price", "photo", "category", "rating"]


class CartItemSerializer(serializers.ModelSerializer):
    class Meta:
        model = CartItem
        fields = ['id', "item", "quantity"]


class CategorySerializer(serializers.ModelSerializer):
    class Meta:
        model = Category
        fields = ['id', "name"]


class RestaurantSerializer(serializers.ModelSerializer):
    class Meta:
        model = Restaurant
        fields = ['id', "name", "address", "rating", "category_id", "image"]


class RestaurantCatSerializer(serializers.ModelSerializer):
    class Meta:
        model = RestaurantCat
        fields = ['id', "name"]
