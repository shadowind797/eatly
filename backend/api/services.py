from django.conf import settings
from django.shortcuts import redirect
from django.core.exceptions import ValidationError
from urllib.parse import urlencode
from typing import Dict, Any
import requests
from .models import User
import jwt

GOOGLE_ACCESS_TOKEN_OBTAIN_URL = 'https://oauth2.googleapis.com/token'
GOOGLE_USER_INFO_URL = 'https://www.googleapis.com/oauth2/v3/userinfo'

GITHUB_ACCESS_TOKEN_OBTAIN_URL = 'https://github.com/login/oauth/access_token'
GITHUB_USER_INFO_URL = 'https://api.github.com/user'

LOGIN_URL = f'{settings.BASE_APP_URL}/login'


def google_get_access_token(code: str, redirect_uri: str) -> str:
    data = {
        'code': code,
        'client_id': settings.GOOGLE_OAUTH2_CLIENT_ID,
        'client_secret': settings.GOOGLE_OAUTH2_CLIENT_SECRET,
        'redirect_uri': redirect_uri,
        'grant_type': 'authorization_code'
    }

    response = requests.post(GOOGLE_ACCESS_TOKEN_OBTAIN_URL, data=data)
    if not response.ok:
        raise ValidationError('Could not get access token from Google.')

    access_token = response.json()['access_token']

    return access_token


def google_get_user_info(access_token: str) -> Dict[str, Any]:
    response = requests.get(
        GOOGLE_USER_INFO_URL,
        params={'access_token': access_token}
    )

    if not response.ok:
        raise ValidationError('Could not get user info from Google.')

    return response.json()


def github_get_access_token(code: str, redirect_uri: str) -> str:
    headers = {
        "Accept": "application/json"
    }
    data = {
        'code': code,
        'client_id': settings.GITHUB_OAUTH2_CLIENT_ID,
        'client_secret': settings.GITHUB_OAUTH2_CLIENT_SECRET,
        'redirect_uri': redirect_uri
    }

    response = requests.post(GITHUB_ACCESS_TOKEN_OBTAIN_URL, headers=headers, data=data)
    if not response.ok:
        raise ValidationError('Could not get access token from Github.')

    access_token = response.json()['access_token']

    return access_token


def github_get_user_info(access_token: str) -> Dict[str, Any]:
    headers = {
        'Authorization': f'Bearer {access_token}'
    }
    response = requests.get(
        GITHUB_USER_INFO_URL,
        headers=headers
    )

    if not response.ok:
        raise ValidationError('Could not get user info from Github.')

    return response.json()


def get_user_data(validated_data, platform):
    domain = settings.BASE_API_URL
    redirect_uri = f'{domain}/api/login/{platform}/'

    code = validated_data.get('code')
    error = validated_data.get('error')
    state = validated_data.get('state')

    if error or not code:
        params = urlencode({'error': error})
        return redirect(f'{LOGIN_URL}?{params}')

    if platform == "google":
        access_token = google_get_access_token(code=code, redirect_uri=redirect_uri)
        user_data = google_get_user_info(access_token=access_token)

        user, created = User.objects.get_or_create(
            email=user_data['email'],
        )

        if not created:
            user.username = user_data['name']
            user.save()

        profile_data = {
            'email': user_data['email'],
        }
        return profile_data
    elif platform == "github":
        access_token = github_get_access_token(code=code, redirect_uri=redirect_uri)
        user_data = github_get_user_info(access_token=access_token)

        print(user_data)

        User.objects.get_or_create(
            username=user_data['login'],
        )

        profile_data = {
            'username': user_data['login'],
        }
        return profile_data
