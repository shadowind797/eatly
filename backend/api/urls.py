from django.urls import path, include

from . import views

urlpatterns = [
    path('access/', views.GetAccess.as_view(), name='access'),
    path('user/', views.GetUser.as_view(), name='user'),
    path('user/change/', views.GetUser.as_view(), name='user-change'),

    path("login/google/", views.GoogleLoginApi.as_view(), name='google-login'),
    path("login/github/", views.GithubLoginApi.as_view(), name='github-login'),

    path('items/', views.ItemListCreate.as_view(), name='item-list'),
    path('items/top/', views.ItemListCreate.as_view(), name='item-list'),
    path('items/category/', views.CategoriesList.as_view(), name='item-category-list'),

    path('items/cart/', views.CartItemListCreate.as_view(), name='item-cart-list'),
    path('items/cart/add', views.CartItemListCreate.as_view(), name='item-cart-list'),
    path('items/cart/delete', views.DeleteCartItem.as_view(), name='item-cart-list'),
    path('items/cart/check', views.CheckInCart.as_view(), name='item-cart-list'),

    path("items/delete/<int:pk>/", views.ItemDelete.as_view(), name='item-delete'),

    path("restaurants/", views.RestaurantListCreate.as_view(), name='restaurant-list'),
    path("restaurants/categories/", views.RestaurantCategories.as_view(), name='restaurant-cats-list'),

    path("address/", views.AddressList.as_view(), name='address-list'),
    path("address/add/", views.AddressList.as_view(), name='restaurant-cats-list'),

    path("order/", views.OrderView.as_view(), name='order'),
    path("order/add/", views.OrderView.as_view(), name='order-create'),
    path("order/cancel/", views.CancelOrder.as_view(), name='order-cancel'),

    path("payment/", views.PaymentView.as_view(), name='payment'),
    path("payment/add/", views.PaymentView.as_view(), name='payment-create'),

    path("coupon/", views.CouponListCreate.as_view(), name='coupon-list'),
    path("coupon/add/", views.CouponListCreate.as_view(), name='coupon-create'),

    path("items/search/", views.SearchView.as_view(), name='search'),
    path("items/search/filters/", views.FilterView.as_view(), name='filter'),

    path("profile/orders", views.GetProfile.as_view(), name='all-user-orders'),
]
