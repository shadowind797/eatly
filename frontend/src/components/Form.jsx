import {useState} from "react";
import api from "../api.js"
import {useNavigate} from "react-router-dom";
import {ACCESS_TOKEN, REFRESH_TOKEN} from "../constants.js";
import LogSlider from "./LogIn/Slider.jsx";

import google from "../assets/google.svg"
import github from "../assets/github.svg"
import hide_pass from "../assets/eye-off.svg"
import show_pass from "../assets/eye-show.svg"
import GoogleButton from "./LogIn/GoogleButton.jsx";

function Form({route, method}) {
    const [username, setUsername] = useState("")
    const [password, setPassword] = useState("")
    const [password2, setPassword2] = useState("")
    const [email, setEmail] = useState("")
    const [phone, setPhone] = useState("")

    const [loading, setLoading] = useState(false)
    const navigate = useNavigate();
    const name = method === "login" ? "Log In" : "Sign Up"
    const [error, setError] = useState("")

    const [passState, setPassState] = useState("password")

    const handleSubmit = async (e) => {
        e.preventDefault()

        if (password === password2 || method === "login") {
            if (email.includes("@") && email.includes(".") || method === "login") {
                if (phone.includes("+") && phone.length > 5 || phone === "") {
                    if (username.length > 0) {
                        if (password.length > 0) {
                            setLoading(true)
                            try {
                                const res = await api.post(route, {username, password, email, phone})
                                if (method === "login") {
                                    localStorage.setItem(ACCESS_TOKEN, res.data.access)
                                    localStorage.setItem(REFRESH_TOKEN, res.data.refresh)
                                    navigate("/")
                                } else {
                                    navigate("/login")
                                }
                            } catch (error) {
                                if (error.response.data.detail) {
                                    setError(error.response.data.detail)
                                } else if (error.response.data.username[0]) {
                                    setError(error.response.data.username[0])
                                }
                            } finally {
                                setLoading(false)
                            }
                        } else {
                            setError("Password is required")
                        }
                    } else {
                        setError("Username is required")
                    }
                } else {
                    setError("Invalid phone number")
                }
            } else {
                setError("Invalid email address")
            }
        } else {
            setError(`Passwords aren't match`)
        }
    }

    const fields = [
        {
            id: 1,
            placeholder: 'USERNAME',
            type: 'text',
            value: username,
            onChange: e => {
                setUsername(e.target.value)
            }
        },
        {
            id: 2,
            placeholder: 'EMAIL',
            type: 'text',
            value: email,
            onChange: e => {
                setEmail(e.target.value)
            }
        },
        {
            id: 3,
            placeholder: 'PHONE (OPTIONAL)',
            type: 'text',
            value: phone,
            onChange: e => {
                setPhone(e.target.value)
            }
        },
        {
            id: 4,
            placeholder: 'PASSWORD',
            type: 'password',
            value: password,
            onChange: e => {
                setPassword(e.target.value)
            }
        },
        {
            id: 5,
            placeholder: 'REPEAT PASSWORD',
            type: 'password',
            value: password2,
            onChange: e => {
                setPassword2(e.target.value)
            }
        },
    ]

    return (
        <div id="log">
            <div id="form">
                <h1>{name} to EATLY</h1>
                <div id="foreign-log">
                    <GoogleButton/>
                    <button id="git-hub"><img src={github} alt=""/></button>
                </div>
                <form onSubmit={handleSubmit}>
                    <div id="inputs">
                        {fields.map(field => {
                            if (method === "login") {
                                if (field.id === 2 || field.id === 3 || field.id === 5) {
                                    return <div key={field.id}></div>
                                } else {
                                    return (
                                        <div id="input" key={field.id}>
                                            <input value={field.value}
                                                   onChange={field.onChange}
                                                   type={field.placeholder === "PASSWORD" ? passState : field.type}
                                                   placeholder={field.placeholder}
                                            />
                                            {field.placeholder === "PASSWORD" &&
                                                <img src={passState === "password" ? show_pass : hide_pass}
                                                     style={{width: "30px", cursor: "pointer"}}
                                                     onClick={() => {
                                                         const passType = passState === "password" ? "text" : "password"
                                                         setPassState(passType)
                                                     }}
                                                     alt=""/>}
                                        </div>
                                    )
                                }
                            } else {
                                return (
                                    <div id="input" key={field.id}>
                                        <input value={field.value}
                                               onChange={field.onChange}
                                               type={field.type}
                                               placeholder={field.placeholder}
                                        />
                                    </div>
                                )
                            }
                        })}
                    </div>
                    <div id="action">
                        <p className="auth-error">{error.length > 0 ? error : ""}</p>
                        <button type="submit">{loading ? `Trying to ${name}...` : name}</button>
                        <p className="another">
                            {method === "login" ? "Don't have an account? " : "Already have an account? "}
                            <a href={`/${method === "login" ? "register" : "login"}`}>
                                {name === "Log In" ? "Sign Up" : "Log In"}
                            </a>
                        </p>
                    </div>
                </form>
                <div id="login-footer">
                    <a href="/privacy-policy">Privacy Policy</a>
                    <p>Made by shadowind in 2024</p>
                </div>
            </div>
            <div id="decor">
                <LogSlider/>
            </div>
        </div>
    )

}

export default Form;
