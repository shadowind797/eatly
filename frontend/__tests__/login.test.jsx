// Form.test.js
import React from 'react';
import {render, fireEvent, waitFor} from '@testing-library/react';
import {useState} from 'react';
import api from '../src/api.js';
import {MemoryRouter as Router} from "react-router-dom";
import Form from '../src/components/Form.jsx';

jest.mock('../src/api');

describe('Form component', () => {
    it('renders without crashing', () => {
        render(<Router><Form route="login" method="login"/></Router>);
    });

    it('matches the snapshot', () => {
        const {asFragment} = render(<Router><Form route="login" method="login"/></Router>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('handles form submission', async () => {
        api.post.mockResolvedValue({
            data: {
                access: 'token123',
                refresh: 'token456',
            },
        });

        const {getByText, getByPlaceholderText} = render(
            <Router><Form route="login" method="login"/></Router>
        );

        const passwordInput = getByPlaceholderText('PASSWORD');
        const usernameInput = getByPlaceholderText('USERNAME');
        const submitButton = getByText('Log In');

        fireEvent.change(passwordInput, {target: {value: 'user123'}});
        fireEvent.change(usernameInput, {target: {value: 'pass123'}});
        fireEvent.click(submitButton);

        await waitFor(() => {
            expect(window.location.pathname).toBe('/');
        });
    });
    it('handles invalid form submission', async () => {
        const {
            getByText,
            getByPlaceholderText
        } = render(
            <Router><Form route="login" method="login"/></Router>
        );

        const passwordInput = getByPlaceholderText('PASSWORD');
        const submitButton = getByText('Log In');

        fireEvent.change(passwordInput, {target: {value: 'password123'}});
        fireEvent.click(submitButton);

        await waitFor(() => {
            expect(getByText('Username is required')
                .textContent).toBe('Username is required')
        });
    });

    it('handles error from API call', async () => {
        api.post.mockRejected({
            response: {
                status: 400,
                data: {
                    detail: "We haven't users with given credentials"
                }
            }
        });

        const {getByText, getByPlaceholderText, getByText: getErrorText} = render(
            <Router><Form route="login" method="login"/></Router>
        );

        const passwordInput = getByPlaceholderText('PASSWORD');
        const password2Input = getByPlaceholderText('USERNAME');
        const submitButton = getByText('Log In');

        fireEvent.change(passwordInput, {target: {value: 'password123'}});
        fireEvent.change(password2Input, {target: {value: 'noname123'}});
        fireEvent.click(submitButton);


        await waitFor(() => {
            expect(getErrorText).toBeInTheDocument('Invalid email address');
        });
    });
});