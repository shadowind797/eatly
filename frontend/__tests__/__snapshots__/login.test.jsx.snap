// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    id="log"
  >
    <div
      id="form"
    >
      <h1>
        Log In to EATLY
      </h1>
      <div
        id="foreign-log"
      >
        <button
          id="google"
        >
          <img
            alt=""
            src="test-file-stub"
          />
        </button>
        <button
          id="git-hub"
        >
          <img
            alt=""
            src="test-file-stub"
          />
        </button>
      </div>
      <form>
        <div
          id="inputs"
        >
          <div
            id="input"
          >
            <input
              placeholder="USERNAME"
              type="text"
              value=""
            />
          </div>
          <div />
          <div />
          <div
            id="input"
          >
            <input
              placeholder="PASSWORD"
              type="password"
              value=""
            />
            <img
              alt=""
              src="test-file-stub"
              style="width: 30px; cursor: pointer;"
            />
          </div>
          <div />
        </div>
        <div
          id="action"
        >
          <p
            class="auth-error"
          />
          <button
            type="submit"
          >
            Log In
          </button>
          <p
            class="another"
          >
            Don't have an account? 
            <a
              href="/register"
            >
              Sign Up
            </a>
          </p>
        </div>
      </form>
      <div
        id="login-footer"
      >
        <a
          href="/privacy-policy"
        >
          Privacy Policy
        </a>
        <p>
          Made by shadowind in 2024
        </p>
      </div>
    </div>
    <div
      id="decor"
    >
      <div
        class="slider_style"
        data-autoplay="false"
        data-infinite="false"
        data-slider="chiefslider"
        id="slider"
      >
        <div
          class="slider__container"
        >
          <div
            class="slider__wrapper"
          >
            <div
              class="slider__items"
            >
              <div
                class="slider__item"
                data-index="0"
                data-order="0"
                data-translate="0"
              >
                <div
                  class="slider__item-container"
                >
                  <div
                    class="slider__item-content"
                  >
                    <div
                      id="second"
                    >
                      <img
                        alt=""
                        class="plate"
                        src="undefined/media/img/food/ChikenHellPro.png"
                      />
                      <div
                        class="bg"
                      />
                      <div
                        class="grafic"
                      >
                        <nav
                          id="up"
                        >
                          <p
                            class="main"
                          >
                            Main
                          </p>
                          <p>
                            Weekly
                          </p>
                          <p>
                            Monthly
                          </p>
                          <p>
                            Yearly
                          </p>
                        </nav>
                        <div>
                          <img
                            alt=""
                            class="g"
                            src="undefined/media/img/vectors/grafic.svg"
                          />
                          <img
                            alt=""
                            class="l"
                            src="undefined/media/img/vectors/lines.svg"
                          />
                          <img
                            alt=""
                            class="p"
                            src="undefined/media/img/vectors/point.svg"
                          />
                        </div>
                        <nav
                          id="down"
                        >
                          <p
                            class="main"
                          >
                            2 days
                          </p>
                          <p>
                            4 days
                          </p>
                          <p>
                            10 days
                          </p>
                        </nav>
                      </div>
                      <div
                        class="chicken-hell"
                      >
                        <div>
                          <img
                            alt=""
                            src="undefined/media/img/food/ChikenHellPro.png"
                          />
                          <div>
                            <h4>
                              Chicken Hell
                            </h4>
                            <h5>
                              On The Way
                            </h5>
                          </div>
                        </div>
                        <p>
                          3:09 PM
                        </p>
                      </div>
                    </div>
                    <h1>
                      Find Foods With Love
                    </h1>
                    <p>
                      Eatly Is The Food Delivery Dashboard And Having More Than 2K+ Dishes Including Asian, Chinese, Italians And Many More. Our Dashboard Helps You To Manage Orders And Money.
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="slider__item"
                data-index="1"
                data-order="1"
                data-translate="0"
              >
                <div
                  class="slider__item-container"
                >
                  <div
                    class="slider__item-content"
                  >
                    <div
                      id="second"
                    >
                      <img
                        alt=""
                        class="plate"
                        src="undefined/media/img/food/ChikenHellPro.png"
                      />
                      <div
                        class="bg"
                      />
                      <div
                        class="grafic"
                      >
                        <nav
                          id="up"
                        >
                          <p
                            class="main"
                          >
                            Main
                          </p>
                          <p>
                            Weekly
                          </p>
                          <p>
                            Monthly
                          </p>
                          <p>
                            Yearly
                          </p>
                        </nav>
                        <div>
                          <img
                            alt=""
                            class="g"
                            src="undefined/media/img/vectors/grafic.svg"
                          />
                          <img
                            alt=""
                            class="l"
                            src="undefined/media/img/vectors/lines.svg"
                          />
                          <img
                            alt=""
                            class="p"
                            src="undefined/media/img/vectors/point.svg"
                          />
                        </div>
                        <nav
                          id="down"
                        >
                          <p
                            class="main"
                          >
                            2 days
                          </p>
                          <p>
                            4 days
                          </p>
                          <p>
                            10 days
                          </p>
                        </nav>
                      </div>
                      <div
                        class="chicken-hell"
                      >
                        <div>
                          <img
                            alt=""
                            src="undefined/media/img/food/ChikenHellPro.png"
                          />
                          <div>
                            <h4>
                              Chicken Hell
                            </h4>
                            <h5>
                              On The Way
                            </h5>
                          </div>
                        </div>
                        <p>
                          3:09 PM
                        </p>
                      </div>
                    </div>
                    <h1>
                      Find Foods With Love
                    </h1>
                    <p>
                      Eatly Is The Food Delivery Dashboard And Having More Than 2K+ Dishes Including Asian, Chinese, Italians And Many More. Our Dashboard Helps You To Manage Orders And Money.
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="slider__item"
                data-index="2"
                data-order="-1"
                data-translate="-300"
                style="transform: translateX(-300%);"
              >
                <div
                  class="slider__item-container"
                >
                  <div
                    class="slider__item-content"
                  >
                    <div
                      id="second"
                    >
                      <img
                        alt=""
                        class="plate"
                        src="undefined/media/img/food/ChikenHellPro.png"
                      />
                      <div
                        class="bg"
                      />
                      <div
                        class="grafic"
                      >
                        <nav
                          id="up"
                        >
                          <p
                            class="main"
                          >
                            Main
                          </p>
                          <p>
                            Weekly
                          </p>
                          <p>
                            Monthly
                          </p>
                          <p>
                            Yearly
                          </p>
                        </nav>
                        <div>
                          <img
                            alt=""
                            class="g"
                            src="undefined/media/img/vectors/grafic.svg"
                          />
                          <img
                            alt=""
                            class="l"
                            src="undefined/media/img/vectors/lines.svg"
                          />
                          <img
                            alt=""
                            class="p"
                            src="undefined/media/img/vectors/point.svg"
                          />
                        </div>
                        <nav
                          id="down"
                        >
                          <p
                            class="main"
                          >
                            2 days
                          </p>
                          <p>
                            4 days
                          </p>
                          <p>
                            10 days
                          </p>
                        </nav>
                      </div>
                      <div
                        class="chicken-hell"
                      >
                        <div>
                          <img
                            alt=""
                            src="undefined/media/img/food/ChikenHellPro.png"
                          />
                          <div>
                            <h4>
                              Chicken Hell
                            </h4>
                            <h5>
                              On The Way
                            </h5>
                          </div>
                        </div>
                        <p>
                          3:09 PM
                        </p>
                      </div>
                    </div>
                    <h1>
                      Find Foods With Love
                    </h1>
                    <p>
                      Eatly Is The Food Delivery Dashboard And Having More Than 2K+ Dishes Including Asian, Chinese, Italians And Many More. Our Dashboard Helps You To Manage Orders And Money.
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <ol
          class="slider__indicators"
        >
          <li
            data-slide-to="0"
          />
          <li
            data-slide-to="1"
          />
          <li
            data-slide-to="2"
          />
        </ol>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
